rust_src/src/editfns.rs

```diff
@@ -1393,4 +1395,95 @@ pub fn delete_and_extract_region(
     }
 }
 
+fn time_arith<F: FnOnce(LispTime, LispTime) -> LispTime>(
+    a: LispObject,
+    b: LispObject,
+    op: F,
+) -> Vec<EmacsInt> {
+    let mut alen: c_int = 0;
+    let mut blen: c_int = 0;
+    let ta = unsafe { lisp_time_struct(a, &mut alen) };
+    let tb = unsafe { lisp_time_struct(b, &mut blen) };
+    let t = op(ta, tb);
+    if LispObject::fixnum_overflow(t.hi) {
+        time_overflow();
+    }
+
+    let maxlen = max(alen, blen) as usize;
+
+    t.into_vec(maxlen)
+}
+
```

Comment:
```
impl Sub for LispTime {
    type Output = LispTime;

    fn sub(self, other: Self) -> Self::Output {
    }
}
```

Along with a match `Add` round out the LispTime. Now you can say `time1 - time2`. If you copy the `time_add` logic into the `add` trait method poof. Done. Then the calls below become `time_arith(a, b, |ta, tb| ta + tb)`.